modules/collections/client/controllers/collections.controllers.js

```diff
@@ -0,0 +1,344 @@
+            self.actionOK = 'Move';
+            self.actionCancel = 'Cancel';
+            self.ok = function() {
+              $uibModalInstance.close($scope.project);
+            };
+            self.cancel = function() {
+              $uibModalInstance.dismiss('cancel');
+            };
+          },
+          controllerAs: 'self',
+          scope: $scope,
+          size: 'md',
+          windowClass: 'modal-alert',
+          backdropClass: 'modal-alert-backdrop'
+        });
+        modalDocView.result.then(function(/* res */) {
+          moveFunc();
+        });
+      };
+
+      var goToList = function() {
+        $state.transitionTo('p.collection.list', { projectid: project.code }, {
+          reload: true, inherit: false, notify: true
+        });
+      };
+
+      var goToDetail = function() {
+        $state.transitionTo('p.collection.edit', { projectid: project.code, collectionId: collection._id }, {
+          reload: true, inherit: false, notify: true
+        });
+      };
+
+      var reloadEdit = function() {
+      // want to reload this screen, do not catch unsaved changes (we are probably in the middle of saving).
+        $scope.allowTransition = true;
+        $state.reload();
+      };
+
+      $scope.otherDocsReordered = function() {
+        reloadEdit();
+      };
+
+      $scope.goToDocument = function(doc) {
+      // Open document in doc manager.
+        $location.url('/p/' + $scope.project.code + '/docs?folder=' + doc.directoryID);
+      };
+
+      $scope.updateDocuments = function(updatedDocuments, sourceDocuments, docType) {
+        var originalDocuments = _.map(sourceDocuments, function(cd) { return cd.document; });
+
+        // Find documents added to the collection
+        var addedDocuments = _.filter(updatedDocuments, function(updatedDoc) {
+          return !_.find(originalDocuments, function(originalDoc) { return originalDoc._id === updatedDoc._id; });
+        });
+
+        // Find documents removed from the collection
+        var removedDocuments = _.filter(originalDocuments, function(originalDoc) {
+          return !_.find(updatedDocuments, function(updatedDoc) { return updatedDoc._id === originalDoc._id; });
+        });
+
+        // Generate promise for document management
+        var docPromises = _.union(_.map(addedDocuments, function(doc) {
+          return CollectionModel.addDocument(collection._id, doc._id, docType);
+        }), _.map(removedDocuments, function(doc) {
+          return CollectionModel.removeDocument(collection._id, doc._id, docType);
+        }));
+
+        // complete the task
+        Promise.all(docPromises)
+          .then(reloadEdit)
+          .catch(function(/* res */) {
+            $scope.showError('Could not update other documents for "'+ $scope.collection.displayName +'".', [], reloadEdit, 'Update Other Documents Error');
+          });
+      };
+
+      $scope.removeDocument = function(document, type) {
```

Comment:
Merged _removeMainDocument_ and _removeOtherDocument_ into one function that flexes based on the type of document being processed (added as a parameter in the function call).